src/containers/sign-on/components/desktop/EmailPage.tsx

```diff
@@ -30,53 +30,78 @@ const errorMessages = {
   inUse: 'Email is already in use, please sign-in'
 }
 
-export class EmailPage extends Component {
-  state = {
-    attempted: false,
-    isSubmitted: false
+type EmailPageProps = {
+  isMobile?: boolean
+  hasMetaMask: boolean
+  email: {
+    value: string
+    status: string
+    error: 'inUse' | 'characters'
   }
+  onNextPage: () => void
+  onEmailChange: (email: string) => void
+  onToggleMetaMaskModal: () => void
+  onSignIn: () => void
+}
+type EmailPageState = {
+  showValidation: boolean
+  isSubmitting: boolean
+}
 
-  onEmailChange = email => {
+export class EmailPage extends Component<EmailPageProps, EmailPageState> {
+  constructor(props: EmailPageProps) {
+    super(props)
+
+    this.state = {
+      showValidation: false,
+      isSubmitting: false
+    }
+  }
+
+  onEmailChange = (email: string) => {
     this.props.onEmailChange(email)
   }
 
   onBlur = () => {
-    this.setState({ attempted: true })
+    this.setState({ showValidation: true })
   }
 
-  onKeyDown = e => {
-    if (e.keyCode === 13 /** enter */) {
+  onKeyDown = (e: KeyboardEvent) => {
+    if (e.key === 'Enter') {
```

Comment:
FYI: `keyCode` is deprecated.

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

not sure if I should be using `key` or `code` here - mdn recommends `code` but other sources recommend using `key` - not sure if it matters.